scipy/sparse/csgraph/_reordering.pyx

```diff
@@ -247,3 +253,45 @@ def _maximum_bipartite_matching(
                     visited[match[queue[j]]] = -1
 
     return match
+
+
+def structural_rank(graph):
+    """
+    structural_rank(graph)
+    
+    Compute the structural rank of a graph (matrix) with a given 
+    sparsity pattern.
+
+    The structural rank of a matrix is the number of entries in the maximum 
+    transversal of the corresponding bipartite graph, and is an upper bound 
+    on the numerical rank of the matrix. A graph has full structural rank 
+    if it is possible to permute the elements to make the diagonal zero-free.
+
+    Parameters
+    ----------
+    graph : sparse matrix
+        Input sparse matrix.
+
+    Returns
+    -------
+    rank : int
+        The structural rank of the sparse graph.
+
```

Comment:
If you have a good reference for this function (I couldn't find one with a quick search), then would be good to add it to the ``References`` section.